com.microsoft.mrtk.input/Utilities/PointingPosePolyfill.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.Subsystems;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// A small PoseDriver-style script that drives the transform of the attached object
+    /// according to a synthesized hand ray vector. A temporary polyfill for devices
+    /// that are missing a pointing pose when using hand tracking.
+    /// </summary>
+    internal class PointingPosePolyfill : MonoBehaviour
+    {
+        private ArticulatedHandController controller;
+        private HandRay handRay;
+        private HandsAggregatorSubsystem handsAggregator;
+        private Pose originalTransform;
+        private bool transformDirty = false;
+
+        private void Awake()
+        {
+            handRay = new HandRay();
+            controller = GetComponentInParent<ArticulatedHandController>();
+            originalTransform = new Pose(transform.localPosition, transform.localRotation);
+        }
+
+        private void OnEnable()
+        {
+            Application.onBeforeRender += OnBeforeRender;
+        }
+
+        private void OnDisable()
+        {
+            Application.onBeforeRender -= OnBeforeRender;
+            transform.localPosition = originalTransform.position;
+            transform.localRotation = originalTransform.rotation;
+        }
+
+        private void OnBeforeRender()
+        {
+            // If the position/pose action on the controller is unbound, we need
+            // to polyfill with a fake pointing pose. (For now, until we have
+            // universal hand interaction profiles.)
+            if (controller?.positionAction.action?.activeControl?.device == null)
```

Comment:
Do you think it'd be safe to cache slightly deeper into this chain? Like, instead of just caching the `ArticulatedHandController`, maybe caching its `positionAction` or `positionAction.action` instead?

Reply:
Fixed